src/modules/SambaAPI.py

```diff
@@ -2,20 +2,24 @@
 import_module('PackageSystem')
 import_module('Package')
 from yast import PackageSystem, Package
+if not PackageSystem.Installed('samba-python3'):
+    if Package.InstallAll(['samba-python3']):
+        from samba.net import Net
+        from samba.credentials import Credentials
+        from samba.dcerpc import nbt
+    else:
+        ycpbuiltins.y2error('Failed to import samba python bindings')
 
 # Global response to net.finddc()
 cldap_ret = None
 
 def net_lookup(domain):
     global cldap_ret
-    if not PackageSystem.Installed('samba-python3'):
-        if not Package.InstallAll(['samba-python3']):
-            return
-    from samba.net import Net
-    from samba.credentials import Credentials
-    from samba.dcerpc import nbt
-    net = Net(Credentials())
-    cldap_ret = net.finddc(domain=domain, flags=(nbt.NBT_SERVER_LDAP | nbt.NBT_SERVER_DS))
+    try:
+        net = Net(Credentials())
+        cldap_ret = net.finddc(domain=domain, flags=(nbt.NBT_SERVER_LDAP | nbt.NBT_SERVER_DS))
+    except NameError:
+        ycpbuiltins.y2error('The samba python bindings are not installed')
```

Comment:
Is the try/except necessary here ? I mean is it possible to throw an error if the package install fails forcing the import to fail thus making sure noone can call net_lookup, it's gonna be kindof ugly if you have to do a try/catch around every imported api statement

Reply:
So, you're suggesting I make the SambaAPI yast import fail somehow instead?

Reply:
yep